core/src/main/java/tech/pegasys/web3signer/core/Runner.java

```diff
@@ -179,51 +168,82 @@ private void registerHttpHostAllowListHandler(final OpenAPI3RouterFactory openAp
 
   private OpenAPI3RouterFactory createOpenApiRouterFactory(
       final Vertx vertx,
-      final KeyIdentifiers keyIdentifiers,
+      final Set<String> blsIdentifiers,
+      final Set<String> secpIdentifiers,
       final MetricsSystem metricsSystem,
       final SignerForIdentifier<BlsArtifactSignature> blsSigner,
       final SignerForIdentifier<SecpArtifactSignature> secpSigner)
       throws InterruptedException, ExecutionException {
     final LogErrorHandler errorHandler = new LogErrorHandler();
     final OpenAPI3RouterFactory openAPI3RouterFactory = getOpenAPI3RouterFactory(vertx);
 
-    // upcheck handler
-    openAPI3RouterFactory.addHandlerByOperationId(
-        UPCHECK_OPERATION_ID, new BlockingHandlerDecorator(new UpcheckHandler(), false));
-    openAPI3RouterFactory.addFailureHandlerByOperationId(UPCHECK_OPERATION_ID, errorHandler);
-
-    // public key handler
-    openAPI3RouterFactory.addHandlerByOperationId(
-        GET_PUBLIC_KEYS_OPERATION_ID,
-        new BlockingHandlerDecorator(new GetPublicKeysHandler(keyIdentifiers), false));
-    openAPI3RouterFactory.addFailureHandlerByOperationId(
-        GET_PUBLIC_KEYS_OPERATION_ID, errorHandler);
+    // public key list handler
+    addEth1PublicKeysListHandler(openAPI3RouterFactory, secpIdentifiers, errorHandler);
+    addEth2PublicKeysListHandler(openAPI3RouterFactory, blsIdentifiers, errorHandler);
 
     // sign handler
-    final Counter missingSignerCounter =
-        metricsSystem.createCounter(
-            Web3SignerMetricCategory.SIGNING,
-            "missing_identifier_count",
-            "Number of signing operations requested, for keys which are not available");
+    addEth1SignHandler(openAPI3RouterFactory, secpSigner, metricsSystem, errorHandler);
+    addEth2SignHandler(openAPI3RouterFactory, blsSigner, metricsSystem, errorHandler);
+
+    // upcheck handler
+    addUpcheckHandler(openAPI3RouterFactory, errorHandler);
+
+    return openAPI3RouterFactory;
+  }
 
+  private void addEth2SignHandler(
+      final OpenAPI3RouterFactory openAPI3RouterFactory,
+      final SignerForIdentifier<BlsArtifactSignature> blsSigner,
+      final MetricsSystem metricsSystem,
+      final LogErrorHandler errorHandler) {
     openAPI3RouterFactory.addHandlerByOperationId(
-        SIGN_FOR_IDENTIFIER_OPERATION_ID,
+        ETH2_SIGN.name(),
         new BlockingHandlerDecorator(
-            new SignForIdentifierHandler(blsSigner, metricsSystem, "bls"), true));
+            new SignForIdentifierHandler(
+                blsSigner, new OpenApiOperationsMetrics(metricsSystem, BLS)),
+            false));
+    openAPI3RouterFactory.addFailureHandlerByOperationId(ETH2_SIGN.name(), errorHandler);
+  }
+
+  private void addEth1SignHandler(
```

Comment:
this feels really similar to the Eth2 function - can it be parameterised and duplication removed?